app/models/update_existing_draft_edition.rb

```diff
@@ -79,4 +95,20 @@ def attributes
   def edition_attributes_from_payload
     payload.slice(*Edition::TOP_LEVEL_FIELDS)
   end
+
+  def publisher_first_published_at
```

Comment:
The problem I have here is that the date field names don't match up with the payload attribute names e.g. the date field is `publisher_first_published_at` and the payload attribute is `first_published_at`.  I think this is correct because the caller shouldn't need to know that the publishing API is treating publisher dates differently to its own dates.  However it does mean that we can't cleanly assign the payload values to the date fields without either removing or adding `publisher_` at some point.  While the methods aren't being re-used, I think at least it does make the relationship clear between the payload attributes and the date fields.  Do you know of a way to better handle the the difference in field and payload names?